cg/services/orders/store_order_services/store_pacbio_order_service.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from datetime import datetime
+
+from cg.constants import DataDelivery, Workflow
+from cg.models.orders.order import OrderIn
+from cg.models.orders.sample_base import StatusEnum
+from cg.services.orders.order_lims_service.order_lims_service import OrderLimsService
+from cg.services.orders.submitters.order_submitter import StoreOrderService
+from cg.store.models import ApplicationVersion, Case, CaseSample, Customer, Sample
+from cg.store.store import Store
+
+LOG = logging.getLogger(__name__)
+
+
+class StorePacBioOrderService(StoreOrderService):
+    """Storing service for PacBio Long Read orders."""
+
+    def __init__(self, status_db: Store, lims_service: OrderLimsService):
+        self.status_db = status_db
+        self.lims = lims_service
+
+    def store_order(self, order: OrderIn) -> dict:
+        """Submit a batch of samples for PacBio Long Read delivery."""
+
+        project_data, lims_map = self.lims.process_lims(lims_order=order, new_samples=order.samples)
+        status_data: dict = self.order_to_status(order)
+        self._fill_in_sample_ids(samples=status_data["samples"], lims_map=lims_map)
+        new_samples = self._store_samples_in_statusdb(
+            customer_id=status_data["customer"],
+            order=status_data["order"],
+            ordered=project_data["date"],
+            ticket_id=order.ticket,
+            samples=status_data["samples"],
+        )
+        return {"project": project_data, "records": new_samples}
+
+    @staticmethod
+    def order_to_status(order: OrderIn) -> dict:
+        """Convert order input to status for PacBio-only orders."""
+        status_data = {
+            "customer": order.customer,
+            "order": order.name,
+            "samples": [
+                {
+                    "application": sample.application,
+                    "capture_kit": sample.capture_kit,
+                    "comment": sample.comment,
+                    "data_analysis": sample.data_analysis,
+                    "data_delivery": sample.data_delivery,
+                    "name": sample.name,
+                    "priority": sample.priority,
+                    "sex": sample.sex,
+                    "tumour": sample.tumour,
+                    "volume": sample.volume,
+                }
+                for sample in order.samples
+            ],
+        }
+        return status_data
+
+    def _store_samples_in_statusdb(
+        self, customer_id: str, order: str, ordered: datetime, ticket_id: str, samples: list[dict]
+    ) -> list[Sample]:
+        """Store PacBio samples in the status database including case links and delivery"""
+        customer: Customer = self.status_db.get_customer_by_internal_id(
+            customer_internal_id=customer_id
+        )
+        new_samples = []
+        case: Case = self.status_db.get_case_by_name_and_customer(
+            customer=customer, case_name=ticket_id
+        )
+        submitted_case: dict = samples[0]
```

Comment:
This line looks very iffy to me. Why are we calling it "submitted case" if we are taking something from the samples dict?

Reply:
renamed